plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/loading/SonarPluginRequirementsChecker.java

```diff
@@ -168,19 +167,28 @@ private static PluginRequirementsCheckResult checkUnsatisfiedPluginDependency(Pl
       // If we evaluate A before B, then A might be wrongly included
       // But I'm not aware of such case in real life.
       if (depInfo == null || depInfo.isSkipped()) {
-        LOG.debug("Plugin '{}' dependency on '{}' is unsatisfied. Skip loading it.", currentResult.getPlugin().getName(), required.getKey());
-        return new PluginRequirementsCheckResult(currentResult.getPlugin(), new SkipReason.UnsatisfiedDependency(required.getKey()));
+        return processUnsatisfiedDependency(currentResult.getPlugin(), required.getKey());
       }
     }
     var basePluginKey = plugin.getBasePlugin();
-    if (basePluginKey != null) {
-      var baseInfo = currentResultsByKey.get(basePluginKey);
-      if (baseInfo == null || baseInfo.isSkipped()) {
-        LOG.debug("Plugin '{}' dependency on '{}' is unsatisfied. Skip loading it.", currentResult.getPlugin().getName(), basePluginKey);
-        return new PluginRequirementsCheckResult(currentResult.getPlugin(), new SkipReason.UnsatisfiedDependency(basePluginKey));
-      }
+    if (basePluginKey != null && checkForPluginSkipped(currentResultsByKey.get(basePluginKey))) {
+      return processUnsatisfiedDependency(currentResult.getPlugin(), basePluginKey);
+    }
+    if (Set.of("dbd", "dbdpythonfrontend").contains(plugin.getKey()) && checkForPluginSkipped(currentResultsByKey.get(Language.PYTHON.getPluginKey()))) {
```

Comment:
I think you can use `DataFlowBugDetection` class here to get a list of DBD plugins to avoid duplicating the hardcoded set.

Reply:
Done, I was hesitant because the class itself is not public but I changed it